snap/pack/pack.go

```diff
@@ -119,9 +119,11 @@ func loadAndValidate(sourceDir string) (*snap.Info, error) {
 	return info, nil
 }
 
-func snapPath(info *snap.Info, targetDir string) string {
-	snapName := fmt.Sprintf("%s_%s_%v.snap", info.InstanceName(), info.Version, debArchitecture(info))
-	if targetDir != "" {
+func snapPath(info *snap.Info, targetDir, snapName string) string {
+	if snapName == "" {
+		snapName = fmt.Sprintf("%s_%s_%v.snap", info.InstanceName(), info.Version, debArchitecture(info))
+	}
+	if targetDir != "" && !filepath.IsAbs(snapName) {
```

Comment:
Should we validate commands and errour out if absolute file path is given together with target path?